src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void growIfArrayIsFull() {
```

Comment:
The method 'growIfArrayIsFull' should be private if it's only used within the class. Also, the name of the method can be simplified to 'grow' since the condition for growing the array is checked within the method.